projects/auth0-angular/src/lib/auth.service.ts

```diff
@@ -28,58 +29,87 @@ import {
   filter,
   takeUntil,
   distinctUntilChanged,
+  catchError,
+  switchMap,
 } from 'rxjs/operators';
 
 import { Auth0ClientService } from './auth.client';
 import { WindowService } from './window';
 import { AbstractNavigator } from './abstract-navigator';
+import { connectableObservableDescriptor } from 'rxjs/internal/observable/ConnectableObservable';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService implements OnDestroy {
   private isLoadingSubject$ = new BehaviorSubject(true);
   private isAuthenticatedSubject$ = new BehaviorSubject(false);
+  private errorSubject$ = new ReplaySubject<Error>(1);
 
   // https://stackoverflow.com/a/41177163
   private ngUnsubscribe$ = new Subject();
 
+  /**
+   * Emits boolean values indicating the loading state of the SDK
+   */
   readonly isLoading$ = this.isLoadingSubject$.asObservable();
 
+  /**
+   * Emits boolean values indicating the authentication state of the user. If `true`, it means a user has authenticated.
+   * This depends on the value of `isLoading$`, so there is no need to manually check the loading state of the SDK.
+   */
   readonly isAuthenticated$ = this.isLoading$.pipe(
     filter((loading) => !loading),
     distinctUntilChanged(),
     concatMap(() => this.isAuthenticatedSubject$)
   );
 
+  /**
+   * Emits details about the authenticated user when `isAuthenticated$` is `true`.
+   */
   readonly user$ = this.isAuthenticated$.pipe(
     filter((authenticated) => authenticated),
     distinctUntilChanged(),
     concatMap(() => this.auth0Client.getUser())
   );
 
+  /**
+   * Emits errors that occur during login, or when checking for an active session on startup.
+   */
+  readonly error$ = this.errorSubject$.asObservable();
+
   constructor(
     @Inject(Auth0ClientService) private auth0Client: Auth0Client,
     @Inject(WindowService) private window: any,
     private navigator: AbstractNavigator
   ) {
     // https://github.com/angular/angular/issues/12631
```

Comment:
I thought the window service disappeared. Why is this still a constructor parameter?

Reply:
Because I haven't yet merged #48. I see it's now approved, so I can merge and rebase.